Index: trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/context/PartialViewContextImpl.java
===================================================================
--- trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/context/PartialViewContextImpl.java	(revision 1161153)
+++ trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/context/PartialViewContextImpl.java	(working copy)
@@ -64,7 +64,8 @@

     _requestType = ReqType.FULL;

-    if (_PARTIAL_AJAX.equals(extContext.getRequestHeaderMap().get(_FACES_REQUEST)))
+    if (_PARTIAL_AJAX.equals(extContext.getRequestHeaderMap().get(_FACES_REQUEST)) ||
+            _PARTIAL_AJAX.equals(extContext.getRequestParameterMap().get(_FACES_REQUEST)))
     {
       // This request was sent with jsf.ajax.request()
       if (extContext.getRequestParameterMap().get(_TRINIDAD_PPR) != null)
Index: trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/share/util/MultipartFormHandler.java
===================================================================
--- trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/share/util/MultipartFormHandler.java	(revision 1161153)
+++ trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/share/util/MultipartFormHandler.java	(working copy)
@@ -225,7 +225,7 @@
     if (!line.startsWith(_boundary))
     {
       // A better exception would be nice.
-      throw new EOFException();
+      throw new IllegalStateException("Boundary information not found. Unable to process upload.");
     }
   }

@@ -388,9 +388,14 @@
     {
       return null;
     }
+    String boundary = contentType.substring(boundaryStart + _BOUNDARY_PARAMETER.length());
+    final int semicolonIndex = boundary.indexOf(";");
+    if (semicolonIndex > -1) {
+        boundary = boundary.substring(0, semicolonIndex);
+    }

     // Boundary always starts with "--"
-    return "--" + contentType.substring(boundaryStart + _BOUNDARY_PARAMETER.length());
+    return "--" + boundary;
   }

   //Reads the ContentType string out of a line of the incoming request
